tsdb/engine/tsm1/compact.go

```diff
@@ -580,11 +580,28 @@ func (c *DefaultPlanner) Plan(lastWrite time.Time) ([]CompactionGroup, int64) {
 	// each generation in descending break once we see a file less than 4.
 	end := 0
 	start := 0
+	lastHighLevelIndex := -1
+
 	for i, g := range generations {
-		if g.level() <= 3 {
+		// Track the last high-level generation
```

Comment:
@davidby-influx this is what I was discussing during standup where I thought it may be not super performant. Previously we would loop through generations and early break if we hit a case where we knew we would end at in our array. Where as now, we need to check everything. I don't really see it being a problem unless we're looping through thousands of generations, which is highly unlikely. I think at most we will see it in the 10's.